### PR TITLE
Fix tracer file type which should be used to decide the output format.

### PR DESCRIPTION
**Pull request by rockindy**

### CODE REVIEW

**Comment by eyck** [comment]
That's a bit difficult as the tracer only take the stem of the file name. This is because the tracer generates 2 files in the standard configurtion: one for signals and one for transactions. Any ideas on how to handle this?

**Comment by rockindy** [comment]
How about to add two functions to set `tx_type` and `sig_type`?
In the constructor of `tracer`, we can set default values for `tx_type` and `sig_type`(e.g., `TEXT` and `FST`).
If users want to change the output format, they can use additional functions.

For example:
```cpp
tracer(std::string const&& name, bool enable_tx, bool enable_sig, sc_core::sc_object* top, sc_core::sc_module_name const& nm);
set_tx_type(file_type tx_type);
set_sig_type(file_type sig_type);
```

**Comment by eyck** [comment]
But this is exactly what ' tracer(std::string const&& name, file_type 
tx_type, file_type sig_type, sc_core::sc_object* top = nullptr)' is 
doing: you provide a name for the trace files and a type for the 
transaction trace (tx_type) and for the type of the signal trace 
(sig_type). Moreover you can change these setting alter on by setting 
the cci parameter tx_trace_type and sig_trace_type.

How is this different from what you propose?

Best

-Eyck

Am 07.08.2024 um 07:32 schrieb rockindy:

File: src/sysc/scc/tracer.cpp [hunk 1]
```diff
@@ -55,7 +55,7 @@ tracer::tracer(std::string const&& name, file_type tx_type, file_type sig_type,
     if(sig_type == ENABLE)
         sig_type = static_cast<file_type>(sig_trace_type.get_value());
     if(sig_type != NONE) {
-        switch(sig_type) {
+        switch(tx_type) {
```
Comment (eyck) [inline src/sysc/scc/tracer.cpp:58]: This logic is wrong as it effectively skips the signal trace type. This way it falls pacl to vcd if selected any of the tx trace type or it falls back to text type transaction trace if any of the signal trace types is selected.

**Review by eyck — COMMENTED** [review]
As far as I can see the current implementation in main & develop allow to select the output formats for signal tracing and transaction tracing independently. Therefore there need to be 2 of the file_type parameters.

**Comment by rockindy** [comment]
Sorry, the first commit was my fault.
Muy original purpose is try to use `configurable_tracer` to output a vcd file.
So the correct way to add a new constructor for `configurable_tracer` to pass `tx_type` and `sig_type` to `tracer`.

**Pull request merged**